src/Xamarin.Android.Build.Tasks/Resources/LayoutBinding.cs

```diff
@@ -0,0 +1,99 @@
+﻿using System;
+
+using Android.App;
+using Android.Views;
+
+namespace Xamarin.Android.Design
+{
+	abstract class LayoutBinding
+	{
+		public delegate Java.Lang.Object OnLayoutItemNotFoundHandler (int resourceId, Type expectedViewType);
+
+		Activity boundActivity;
+		View boundView;
+		OnLayoutItemNotFoundHandler onLayoutItemNotFound;
+
+		public abstract int ResourceLayoutID { get; }
+
+		protected LayoutBinding (Activity activity, OnLayoutItemNotFoundHandler onLayoutItemNotFound = null)
+		{
+			boundActivity = activity ?? throw new ArgumentNullException (nameof (activity));
+			this.onLayoutItemNotFound = onLayoutItemNotFound;
+		}
+
+		protected LayoutBinding (View view, OnLayoutItemNotFoundHandler onLayoutItemNotFound = null)
+		{
+			boundView = view ?? throw new ArgumentNullException (nameof (view));
+			this.onLayoutItemNotFound = onLayoutItemNotFound;
+		}
+
+		protected T FindView <T> (int resourceId, ref T cachedField) where T: View
+		{
+			if (cachedField != null)
+				return cachedField;
+
+			T ret;
+			if (boundActivity != null)
+				ret = boundActivity.FindViewById <T> (resourceId);
+			else
+				ret = boundView.FindViewById <T> (resourceId);
+
+			if (ret == null && onLayoutItemNotFound != null)
+				ret = (T)onLayoutItemNotFound (resourceId, typeof (T));
+
+			if (ret == null)
+				throw new global::System.InvalidOperationException ($"View not found (Layout ID: {ResourceLayoutID}; Resource ID: {resourceId})");
+
+			cachedField = ret;
+			return ret;
+		}
+
+		Activity EnsureActivity ()
+		{
+			if (boundActivity != null)
+				return boundActivity;
+
+			var ret = boundView?.Context as Activity;
+			if (ret != null)
+				return ret;
+
+			throw new InvalidOperationException ("Finding fragments is supported only for Activity instances");
+		}
+
+		T __FindFragment<T> (int resourceId, Func<FragmentManager, T> finder, ref T cachedField) where T: Java.Lang.Object
```

Comment:
The previous version where `Android.App.Fragment` & the support Fragment `FindFragment()` methods were separate was more correct, because the support library [*provides its own `FragmentManager` type](https://developer.android.com/reference/android/support/v4/app/FragmentManager). Thus, `Func<FragmentManager, T>` doesn't make sense, as there's only one type which can use it.